### PR TITLE
feat: startPos allows specifying the position to tail from

### PR DESCRIPTION
**Pull request by darinspivey**
This feature will allow things like being able to tail a
file from the beginning (by passing 0). If `startPos` is
provided, the tailing will begin at that spot in the file.

Semver: minor
Fixes: https://github.com/logdna/tail-file-node/issues/7
Ref: LOG-7789

### CODE REVIEW

File: lib/tail-file.js [hunk 1]
```diff
@@ -68,6 +69,24 @@ class TailFile extends Readable {
       }
       throw err
     }
+    if (startPos !== null && startPos !== undefined) {
+      if (typeof startPos !== 'number') {
+        const err = new TypeError('startPos must be an integer > 0')
```
Comment (darinspivey) [inline lib/tail-file.js:74]: >=

File: lib/tail-file.js [hunk 2]
```diff
@@ -68,6 +69,24 @@ class TailFile extends Readable {
...
+      }
+      if (startPos < 0 || !Number.isInteger(startPos)) {
+        const err = new TypeError('startPos must be an integer > 0')
+        err.code = 'ESTARTPOS'
+        err.meta = {
+          got: startPos
+        }
+        throw err
+      }
+    }
```
Comment (jmoses) [inline lib/tail-file.js:89]: Can these just be parts of the same if? they do the same thing.

```javascript
if(typeof startpos !== 'number' || (startpos < 0 ....) )
```

Comment (darinspivey) [reply lib/tail-file.js:89]: they're slightly different, enough to warrant them being separate.  One is a type check, and the other is a value check, and the `meta` value is different from both.  It's simpler to read separately, even though they appear to be the same.

Comment (jmoses) [reply lib/tail-file.js:89]: Ah, there we go, I missed the `meta` difference.

File: lib/tail-file.js [hunk 3]
```diff
@@ -68,6 +69,24 @@ class TailFile extends Readable {
...
+      if (typeof startPos !== 'number') {
+        const err = new TypeError('startPos must be an integer >= 0')
+        err.code = 'ESTARTPOS'
+        err.meta = {
+          got: typeof startPos
+        }
+        throw err
+      }
+      if (startPos < 0 || !Number.isInteger(startPos)) {
+        const err = new TypeError('startPos must be an integer >= 0')
```
Comment (mdeltito) [inline lib/tail-file.js:82]: `RangeError` or just `Error`? I think `RangeError` is appropriate for when a positive non-integer value is provided. But doesn't seem like this should be `TypeError`.

> The RangeError object indicates an error when a value is **not in the set** or range of allowed values.

Comment (darinspivey) [reply lib/tail-file.js:82]: yep, that works

**Review by mdeltito — APPROVED** [review]

**Review by evanlucas — APPROVED** [review]

**Pull request merged**